lib/signs/realtime.ex

```diff
@@ -142,12 +142,27 @@ defmodule Signs.Realtime do
   end
 
   def handle_info(:run_loop, sign) do
-    sign_stop_ids = SourceConfig.sign_stop_ids(sign.source_config)
-    sign_routes = SourceConfig.sign_routes(sign.source_config)
-    alert_status = sign.alerts_engine.max_stop_status(sign_stop_ids, sign_routes)
     sign_config = sign.config_engine.sign_config(sign.id)
     current_time = sign.current_time_fn.()
 
+    alert_status =
+      case sign.source_config do
+        {top, bottom} ->
+          top_alert_status = get_alert_status_for_source_config(top, sign.alerts_engine)
+          bottom_alert_status = get_alert_status_for_source_config(bottom, sign.alerts_engine)
+
+          {top_alert_status, bottom_alert_status}
+
+        source_config ->
+          alert_status = get_alert_status_for_source_config(source_config, sign.alerts_engine)
+
+          alert_status
```

Comment:
Doesn't seem like the `alert_status` variable is needed for anything in this scope, unless it's just a preference?

Reply:
Yeah I suppose this is just preference on my part. This produces parallel structure across the two `case` clauses.

The structure being:
1. Get alert status(es)
2. Return alert status(es)

Which is pretty obvious for a single configuration

https://github.com/mbta/realtime_signs/blob/836b0f0cf0e53ba5a23e1fdcd65b93e0a96f0967/lib/signs/realtime.ex#L157-L159

But requires the additional variables for separate top and bottom configurations

https://github.com/mbta/realtime_signs/blob/836b0f0cf0e53ba5a23e1fdcd65b93e0a96f0967/lib/signs/realtime.ex#L151-L154

Removing the intermediate variable for a single configuration would remove the parallel structure:

```elixir
case sign.source_config do
  {top, bottom} ->
    top_alert_status = get_alert_status_for_source_config(top, sign.alerts_engine)
    bottom_alert_status = get_alert_status_for_source_config(bottom, sign.alerts_engine)

    {top_alert_status, bottom_alert_status}

  source_config ->
    get_alert_status_for_source_config(source_config, sign.alerts_engine)
end
```

---

<details>
<summary>(bonus content)</summary>

While on the subject, I think a way we would simplify _a lot_ of this code is with something like `SourceConfig.map/2` (and maybe `SourceConfig.simplify/1`) which would look something like this: https://github.com/mbta/realtime_signs/pull/798/files


</details>